internal/querycoordv2/meta/collection_manager.go

```diff
@@ -188,10 +188,17 @@ func (m *CollectionManager) Recover(broker Broker) error {
 
 // upgradeRecover recovers from old version <= 2.2.x for compatibility.
 func (m *CollectionManager) upgradeRecover(broker Broker) error {
+	// for loaded collection from 2.2, it only save a old version CollectionLoadInfo without LoadType.
+	// we should update the CollectionLoadInfo and save all PartitionLoadInfo to meta store
 	for _, collection := range m.GetAllCollections() {
-		// It's a workaround to check if it is old CollectionLoadInfo because there's no
-		// loadType in old version, maybe we should use version instead.
 		if collection.GetLoadType() == querypb.LoadType_UnKnownType {
+			newInfo := collection.Clone()
+			newInfo.LoadType = querypb.LoadType_LoadCollection
+			err := m.putCollection(true, newInfo)
+			if err != nil {
+				return err
+			}
+
 			partitionIDs, err := broker.GetPartitions(context.Background(), collection.GetCollectionID())
 			if err != nil {
 				return err
```

Comment:
This operation is not atomic here. What will happen if querycoord crash after collection info updated but partitions are not yet?